### PR TITLE
Adds support for camp-style dsl in TOSCA blueprints

### PR DESCRIPTION
**Pull request by nakomis**

### CODE REVIEW

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java [hunk 1]
```diff
@@ -95,6 +105,8 @@ public ToscaNodeToEntityConverter setNodeId(String nodeId) {
         // Add TOSCA node type as a property
         spec.configure("tosca.node.type", this.nodeTemplate.getType());
         spec.configure("tosca.template.id", this.nodeId);
+        // Use the nodeId as the camp.template.id to enable DSL lookup
+        spec.configure(BrooklynCampConstants.PLAN_ID, this.nodeId);
```
Comment (kiuby88) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java:109]: Probably, we should to decide a generic ID for represent plan ids, avoiding "TOSCA" or "CAMP" string. Or adding "tosca.template.id" [here](https://github.com/apache/incubator-brooklyn/blob/master/usage/camp/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/dsl/methods/DslComponent.java#L127)

File: brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java [hunk 2]
```diff
@@ -139,9 +151,9 @@ public ToscaNodeToEntityConverter setNodeId(String nodeId) {
     }
 
     //TODO: refactor this method in Brooklyn in {@link org.apache.brooklyn.camp.brooklyn.spi.creation.BrooklynComponentTemplateResolver}
```
Comment (kiuby88) [inline brooklyn-tosca-transformer/src/main/java/org/apache/brooklyn/tosca/a4c/brooklyn/ToscaNodeToEntityConverter.java:153]: This TODO should be deleted

**Comment by sjcorbett** [comment]
Looks good. Would it take much effort to add test cases?

**Comment by kiuby88** [comment]
Thanks a lot @nakomis, it will be ver useful for ChatApplication example.
I have created #16 which is based on this commit for testing ChatApplication. I have adapted the DSL management for supporting DSLs inside of a Map of properties. @nakomis, I do not want to interfere, so if it was necessary, #16 would be closed.

**Comment by sjcorbett** [comment]
I will merge this since several other pull requests are waiting on it.

**Pull request merged**